algorithms/C/linked-lists/Insert-and-del-beginning-circular-ll.c

```diff
@@ -0,0 +1,92 @@
+
+//Circular Linked List
+//insertion and Deletion at the beginning of the Linked List
+#include<stdio.h>
+#include<stdlib.h>
+
+//  Structure Definition
+
+// 1. Data of Integer type.
+// 2. Pointer to the Next Node .
+struct node {
+    int data ;
+    struct node *next ;
+} ;
+struct node *head = NULL ;
+//Use tail pointer to avoid traversing the Linked List while inserting of the new created node
+//because after every insertion of a new node we need to point make the last node point to the newly created Node 
+struct node *tail = NULL ;
+
+//Function to Print the Linked List
+void printList(){
+    printf("\nItem in the Linked List are :  ");
+    // Initialize the ptr(pointer) with head location .
+    struct node *ptr = head ;
+    do{
+       printf("%d ",ptr->data) ;  //Print the value of the Node
+       ptr = ptr->next ;         //Increment the ptr 
+    }while(ptr!=head);           //// While Loop until we encounter a Node pointer ptr not pointing to the same location as head .
+}
+
+//Insertion at the Beginning of the Circular Linked List
+
+void insertBeg(int data){
+ //Creating a link
+ //Malloc is for space allocation .
+ struct node *new_node = (struct node*) malloc(sizeof(struct node)) ;
+ if(new_node==NULL){
+     printf("Cannot create the Node");
+ }
+ else{
+     
+     new_node->data=data;
+     if(head==NULL){
+     	//making the head and the tail pointer to point to the newly created node
+         head = new_node ;
+         tail = new_node;
+     }
+     else{
+        //Point the new_node pointer to the current head
+        new_node->next = head ;
+        head = new_node ;
+     }
+     //making the tail pointer to point to the new head
+    tail->next=head;
+     
+     
+ }
+ 
+}
+
+//Deletion at the Beginning of the Circular Linked List
+
+void deleteBeg(){
+  //NO Deletion when list is empty
+   if(head == NULL) {
+    printf("List is Empty , deletion not possible") ;
+   }
```

Comment:
you would not want to continue the execution of the function in case the list is empty. 

```suggestion
   if(head == NULL) {
    printf("List is Empty , deletion not possible") ;
    return;
   }
```

Reply:
sir how can i return when the return type of the delete function is void
maybe i should use the exit statement is this fine??

Reply:
True void function cannot return any value but we can use return statement here to terminate the functions execution.

Reply:
Oh ok sir understood i will do that suggested change
Thank you!